### ISSUE TITLE
Confused Properties in 1.1.9

### ISSUE THREAD

**Issue by semiapies** [less/less.ruby#52]
This input:

```
td {
    margin:0;
    padding:0;
    border:0;
    outline:0;
    font-size:9pt;
    vertical-align:baseline;
    background:transparent;
}


td, input {
    line-height:1em;
}   
```

will produce this odd (and very incorrect) output:

```
td, input {
  margin: 0;
  padding: 0;
  border: 0;
  outline: 0;
  font-size: 9pt;
  vertical-align: baseline;
  background: transparent;
}
td { line-height: 1em; }
```

**Comment by cloudhead** [comment]
fixed and refactored selector grouping algorithm. Closed by 6ff094632f98652acd5cb7687b36b8bd959bcab5. Element no longer inherits from String.

**Comment by cloudhead** [comment]
Fixed in 1.1.10 — nasty bug, due to the way equality between two selectors was checked, good find!

**Comment by semiapies** [comment]
Thanks - though I really just reduced that example down from the Meyers reset block when I noticed my input borders and such went away after updating to 1.1.9.  

The fix works very nicely!
